app/lib/application_extension.rb

```diff
@@ -16,6 +16,8 @@ module ApplicationExtension
     validates :redirect_uri, length: { maximum: APP_REDIRECT_URI_LIMIT }
     validates :website, url: true, length: { maximum: APP_WEBSITE_LIMIT }, if: :website?
 
+    normalizes :scopes, with: ->(scopes) { Array(scopes)&.compact&.join(' ') }
```

Comment:
I started a branch to do this same thing, but ran into some weirdness trying to write normalizes specs ... I think because of the custom getter/setter from doorkeeper - https://github.com/doorkeeper-gem/doorkeeper/blob/v5.8.1/lib/doorkeeper/models/concerns/scopes.rb - it was not working as expected.

Was contemplating whether embracing/accepting the custom getter and asseting on what it builds -vs- asserting on `attributes_before_type_cast` or something (to verify that our inbound normalizations do what we want) instead...

Reply:
Oh, I missed that Doorkeeper was doing that. Does that mean that… we don't actually have to process it at all in the controller? :thinking:

The current tests seem to pass when removing the conversion code in the controller.

Reply:
I think that might be the case (that we can just remove custom thing). Looks like the setter was added in 2019 - https://github.com/doorkeeper-gem/doorkeeper/commit/421be784fcdb13e7547459a92f84a260ce8a62dd - and our custom handling from 2017 - https://github.com/mastodon/mastodon/pull/4671 - so seems likely that the support was added during an upgrade but we didnt notice?